### PR TITLE
add optional parameter to support hessian protocol method overload and request protocol version

### PR DESCRIPTION
**Pull request by ma-xiao-guang-64**
add hessian.overload.method to support method overload default is false
add hessian2.request to support request protocol version default is version 1

### CODE REVIEW

**Comment by kimmking** [comment]
good job

File: dubbo-rpc/dubbo-rpc-hessian/src/main/java/com/alibaba/dubbo/rpc/protocol/hessian/HessianProtocol.java [hunk 1]
```diff
@@ -83,6 +83,7 @@ public void run() {
     @SuppressWarnings("unchecked")
     protected <T> T doRefer(Class<T> serviceType, URL url) throws RpcException {
         HessianProxyFactory hessianProxyFactory = new HessianProxyFactory();
+        hessianProxyFactory.setOverloadEnabled(true);
```
Comment (kimmking) [inline dubbo-rpc/dubbo-rpc-hessian/src/main/java/com/alibaba/dubbo/rpc/protocol/hessian/HessianProtocol.java]: I suggest this option can be a optional parameter for hessian protocol, then developers can turn  it on/off.

**Comment by ralf0131** [comment]
Hi, Could you please update the description in your pull request, and your commit message into English?  Dubbo is a global community, please be aware that non-English commit message may not be understood by other people.

File: dubbo-rpc/dubbo-rpc-hessian/src/main/java/com/alibaba/dubbo/rpc/protocol/hessian/HessianProtocol.java (hunk 1 above)
Comment (ma-xiao-guang-64) [reply dubbo-rpc/dubbo-rpc-hessian/src/main/java/com/alibaba/dubbo/rpc/protocol/hessian/HessianProtocol.java]: 已经改成可配置的了   另外加了hessian协议版本的配置  麻烦再review一下

Comment (ma-xiao-guang-64) [reply dubbo-rpc/dubbo-rpc-hessian/src/main/java/com/alibaba/dubbo/rpc/protocol/hessian/HessianProtocol.java]: what happend in jdk8 travis-ci, please help me

File: dubbo-rpc/dubbo-rpc-hessian/src/main/java/com/alibaba/dubbo/rpc/protocol/hessian/HessianProtocol.java [hunk 2]
```diff
@@ -83,6 +83,10 @@ public void run() {
     @SuppressWarnings("unchecked")
     protected <T> T doRefer(Class<T> serviceType, URL url) throws RpcException {
         HessianProxyFactory hessianProxyFactory = new HessianProxyFactory();
+        boolean isHessian2Request = url.getParameter(Constants.HESSIAN2_REQUEST_KEY, Constants.DEFAULT_HESSIAN2_REQUEST);
```
[the remark this answers is not in the document]
Comment (ma-xiao-guang-64) [reply dubbo-rpc/dubbo-rpc-hessian/src/main/java/com/alibaba/dubbo/rpc/protocol/hessian/HessianProtocol.java]: yes, when you add dubbo.service.hessian2.request=true to url the hessian protocal will using hessian 2.
l very agree with your suggest that Constants.HESSIAN2_REQUEST_KEY named dubbo.service.hessian2.request, can i fix it?

File: dubbo-config/dubbo-config-spring/src/main/resources/META-INF/dubbo.xsd [hunk 3]
```diff
@@ -307,6 +317,16 @@
...
+                <xsd:attribute name="hessian2.request" type="xsd:string" use="optional">
+                    <xsd:annotation>
+                        <xsd:documentation><![CDATA[ request use hessian2 protocol. ]]></xsd:documentation>
+                    </xsd:annotation>
+                </xsd:attribute>
+                <xsd:attribute name="hessian.overload.method" type="xsd:string" use="optional">
+                    <xsd:annotation>
+                        <xsd:documentation><![CDATA[ hessian protocol support overload method. ]]></xsd:documentation>
+                    </xsd:annotation>
+                </xsd:attribute>
```
Comment (chickenlj) [inline dubbo-config/dubbo-config-spring/src/main/resources/META-INF/dubbo.xsd]: Why do you put different prefixes `hessian2.request` and `hessian.overload.method`. 
Actually, i think we don't have to add xml support for every parameter, especially for parameters bind to a specific protocol, we can use `properties` or `<dubbo:parameter />` to do that. Your previous changes is ready to work, would you please revert changes adding xml, sorry for misleading you.

**Comment by ma-xiao-guang-64** [comment]
@chickenlj 
ok, i fix it.
now, we can use <dubbo:parameter key="hessian2.request" value="true" /> and <dubbo:parameter key="hessian.overload.method" value="true" />

**Pull request merged**